webui/react/src/components/MetricBadgeTag.tsx

```diff
@@ -11,7 +11,7 @@ interface Props {
 const MetricBadgeTag: React.FC<Props> = ({ metric }: Props) => {
   return (
     <BadgeTag label={metric.name} tooltip={metric.group}>
-      {metric.group.substring(0, 1).toUpperCase()}
+      {(metric.group ?? '').substring(0, 1).toUpperCase()}
```

Comment:
suggestion: 

https://github.com/determined-ai/determined/blob/108ffea29c6d49151012edda042569368d4f8ecb/webui/react/src/types.ts#L474

if it makes sense, i think `group` type should be optional since it is confusing why nullish coalescing operator is here, and it can prevent the future unexpected error

Reply:
I will add a TODO, but the correct thing to do is probably properly decode the metric settings wherever applicable (and throw away any invalid settings). The group should always be defined from the API so I hesitate to make it optional due to settings.